src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -257,7 +258,12 @@ def _run_new_describe(
             and listener.include_describe,
             self._query_listener,
         ):
-            listener._add_query(QueryRecord(cursor.sfqid, query, True))
+            query_record = QueryRecord(cursor.sfqid, query, True)
+            if listener.include_thread_id:
```

Comment:
```suggestion
            if getattr(listener, "include_thread_id", False):
```
so that we don't break custom query listeners.